Library/Homebrew/formula.rb

```diff
@@ -540,8 +540,8 @@ def oldname
   # Old names for the formula.
   sig { returns(T::Array[String]) }
   def oldnames
-    @oldnames ||= if tap
-      T.must(tap).formula_oldnames.fetch(name, [])
+    @oldnames ||= if (tap = self.tap)
+      tap.formula_oldnames.fetch(name, [])
     else
       []
     end
```

Comment:
wdyt of:
```suggestion
    @oldnames ||= tap&.formula_oldnames&.fetch(name, []) || []
```
(Or maybe `&.[](name)` if you want to code golf it)
?

Reply:
If I really wanted to golf it:

```suggestion
    @oldnames ||= [*tap&.formula_oldnames&.[](name)]
```

😂

Reply:
```suggestion
    @oldnames ||= (tap&.formula_oldnames || {}).fetch(name, [])
```

maybe?

Reply:
```suggestion
    @oldnames ||= tap&.formula_oldnames&.dig(name) || []
```

is probably the most readable I can come up with.

Reply:
I'm pretty sure the single method dig will unfortunately not pass Rubocop by default. We could disable that rule though. I like the last one best but I don't really care that much.